### DISCUSSION TITLE
Is there a way to disable the dark mode fade transition?

### DISCUSSION THREAD

**Discussion by TaufiqSyed** [chakra-ui/chakra-ui#4185] [Q&A]
The transition looks pretty jarring when relatively lighter elements transition to relatively darker elements. It just looks kind of jarring. Other sites with famous dark themes like Reddit and Twitter don't seem to have any fade animation. Is there any way to disable the fade transition?

**Comment by mattwells19** [comment]
Yea it's a pretty hot topic that's being investigated by at least one lad here: #3727

**Reply by TaufiqSyed** [reply]
Appreciate the link.

**Reply by TaufiqSyed** [reply]
I'm speaking about disabling the fade transition when toggling dark mode, rather than about the white flash. Sorry for explaining it kinda badly.

**Reply by mattwells19** [reply]
It looks like there's a css variable for transitioning common properties on elements.
![image](https://user-images.githubusercontent.com/36650048/121180468-e589db80-c82e-11eb-86b3-7b004dfaad00.png)

Maybe you're seeing that transition play out when you switch themes. Especially if the colors are very different like you said.

I'm not really sure what the best way is to disable that assuming you only want to disable it when switching themes. You could try setting that css variable to empty when the theme-switch button is clicked, switch the theme, and then reset the css variable. Not sure how well that'll actually work.

Side note: it looks like the body doesn't use this common css variable. It just has it's own background-color transition on the body element so you'll have to edit that one separately if you wanted to.

**Reply by TaufiqSyed** [reply]
> It looks like there's a css variable for transitioning common properties on elements.
> ![image](https://user-images.githubusercontent.com/36650048/121180468-e589db80-c82e-11eb-86b3-7b004dfaad00.png)
> 
> Maybe you're seeing that transition play out when you switch themes. Especially if the colors are very different like you said.
> 
> I'm not really sure what the best way is to disable that assuming you only want to disable it when switching themes. You could try setting that css variable to empty when the theme-switch button is clicked, switch the theme, and then reset the css variable. Not sure how well that'll actually work.
> 
> Side note: it looks like the body doesn't use this common css variable. It just has it's own background-color transition on the body element so you'll have to edit that one separately if you wanted to.

Thanks a lot for the amazing lead, but it goes a little beyond what I'm willing to put in for the weird jerky animation, as well as beyond what I know. Might come back to this if I use this in production.

**Reply by TaufiqSyed** [reply]
> It looks like there's a css variable for transitioning common properties on elements.
> ![image](https://user-images.githubusercontent.com/36650048/121180468-e589db80-c82e-11eb-86b3-7b004dfaad00.png)
> 
> Maybe you're seeing that transition play out when you switch themes. Especially if the colors are very different like you said.
> 
> I'm not really sure what the best way is to disable that assuming you only want to disable it when switching themes. You could try setting that css variable to empty when the theme-switch button is clicked, switch the theme, and then reset the css variable. Not sure how well that'll actually work.
> 
> Side note: it looks like the body doesn't use this common css variable. It just has it's own background-color transition on the body element so you'll have to edit that one separately if you wanted to.

I think I get how to do it now. The delay was probably because it was using the default behaviour, but when I change the element directly using useColorMode it is instant. Saw this used in the examples folder.
